docker-java-core/src/main/java/com/github/dockerjava/core/exec/SearchImagesCmdExec.java

```diff
@@ -23,11 +23,22 @@ public SearchImagesCmdExec(WebTarget baseResource, DockerClientConfig dockerClie
 
     @Override
     protected List<SearchItem> execute(SearchImagesCmd command) {
-        WebTarget webResource = getBaseResource().path("/images/search").queryParam("term", command.getTerm());
+        WebTarget webResource = null;
+
+        Integer limit = command.getLimit();
+        if (limit == null) {
+            webResource = getBaseResource()
+                .path("/images/search")
+                .queryParam("term", command.getTerm());
+        } else {
+            webResource = getBaseResource()
+                .path("/images/search")
+                .queryParam("term", command.getTerm())
+                .queryParam("limit", command.getLimit());
```

Comment:
you can just add if check without duplication
```
        if (nonNull(limit)) {
            webResource = webResource.queryParam("limit", command.getLimit());
        }   
```

Reply:
thanks @KostyaSha, have made change